opal/mca/btl/tcp/btl_tcp.c

```diff
@@ -531,3 +533,64 @@ void mca_btl_tcp_dump(struct mca_btl_base_module_t* base_btl,
     }
 #endif /* OPAL_ENABLE_DEBUG && WANT_PEER_DUMP */
 }
+
+
+/*
+ * A blocking recv on a non-blocking socket. Used to receive the small
+ * amount of connection information that identifies the endpoints
+ * endpoint.
+ */
+
```

Comment:
Argh, I forgot about the second path from the connect_ack. In the case where the socket is in non-blocking mode the timeout attached to the socket will not be useful. We need to fix this in such a way to cover both cases.